### PR TITLE
internal review

### PR DESCRIPTION
**Pull request by jasonf20**

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/connector/CatalogServiceProviderModule.java [hunk 1]
```diff
@@ -85,6 +87,20 @@ public static CatalogServiceProvider<ConnectorNodePartitioningProvider> createNo
...
+    @Provides
+    @Singleton
+    public static CatalogServiceProvider<Optional<ConnectorDynamicFilterProvider>> createDynamicFilterProvider(ConnectorServicesProvider connectorServicesProvider)
+    {
+        return new ConnectorCatalogServiceProvider<>("dynamic filter provider", connectorServicesProvider, ConnectorServices::getDynamicFilterProvider);
+    }
+
+    @Provides
+    @Singleton
+    public static CatalogServiceProvider<Iterable<EventListener>> createEventListeners(ConnectorServicesProvider connectorServicesProvider)
```
Comment (jasonf20) [inline core/trino-main/src/main/java/io/trino/connector/CatalogServiceProviderModule.java]: delete this.

File: core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorDynamicFilterProvider.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    {
+    }
+
+    default void queryCompleted(QueryId queryId)
+    {
+    }
+
+    default DynamicFilter getDynamicFilter(DynamicFilter baseFilter, QueryId queryId)
+    {
+        throw new UnsupportedOperationException();
```
Comment (jasonf20) [inline core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorDynamicFilterProvider.java]: return base.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConnector.java [hunk 3]
```diff
@@ -88,7 +94,8 @@ public IcebergConnector(
             Set<Procedure> procedures,
             Set<TableProcedureMetadata> tableProcedures,
             Set<ConnectorTableFunction> tableFunctions,
-            FunctionProvider functionProvider)
+            FunctionProvider functionProvider,
+            EventListener eventListener)
```
Comment (jasonf20) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConnector.java:98]: remove

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergDynamicFilterProvider.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        synchronized (deleteManagers) {
+            deleteManagers.remove(queryId);
+        }
+    }
+
+    private DeleteManager getDeleteManager(QueryId queryId)
+    {
+        synchronized (deleteManagers) {
+            if (!deleteManagers.containsKey(queryId)){
```
Comment (jasonf20) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergDynamicFilterProvider.java]: get and check if null

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPlugin.java [hunk 5]
```diff
@@ -15,21 +15,44 @@
...
 import io.trino.spi.connector.ConnectorFactory;
+import io.trino.spi.eventlistener.EventListener;
+import io.trino.spi.eventlistener.EventListenerFactory;
 
 import java.util.Set;
 
 public class IcebergPlugin
         implements Plugin
 {
+//    private final Module loggerModule;
```
Comment (jasonf20) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPlugin.java]: delete

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/InternalIcebergConnectorFactory.java [hunk 6]
```diff
@@ -128,6 +133,8 @@ public static Connector createConnector(
                     .flatMap(Collection::stream)
                     .collect(toImmutableList());
 
+            EventListener eventListener = injector.getInstance(EventListener.class);
```
Comment (jasonf20) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/InternalIcebergConnectorFactory.java:136]: delete

File: testing/trino-server-dev/etc/catalog/iceberg.properties [hunk 7]
```diff
@@ -9,9 +9,12 @@ connector.name=iceberg
...
+#
+## Fail-fast in development
+#hive.metastore.thrift.client.max-retry-time=1s
+#iceberg.register-table-procedure.enabled=true
 
-# Fail-fast in development
-hive.metastore.thrift.client.max-retry-time=1s
-iceberg.register-table-procedure.enabled=true
+iceberg.catalog.type=glue
+hive.metastore.glue.region=us-east-1
```
Comment (jasonf20) [inline testing/trino-server-dev/etc/catalog/iceberg.properties]: delete

File: core/trino-main/src/main/java/io/trino/operator/ScanFilterAndProjectOperator.java [hunk 8]
```diff
@@ -99,9 +102,11 @@ private ScanFilterAndProjectOperator(
     {
         pages = splits.flatTransform(
                 new SplitToPages(
-                        session,
+                        operatorContext.getSession(),
```
Comment (yoniiny) [inline core/trino-main/src/main/java/io/trino/operator/ScanFilterAndProjectOperator.java]: move inwards

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/StatefulIcebergPageSourceProvider.java [hunk 9]
```diff
@@ -369,6 +370,8 @@ else if (identity.getId() == TRINO_MERGE_PARTITION_DATA) {
                 .map(readerColumns -> readerColumns.get().stream().map(IcebergColumnHandle.class::cast).collect(toList()))
                 .orElse(requiredColumns);
 
+        DeleteManager deleteManager = getDeleteManager(partitionSpec, partitionData);
```
Comment (yoniiny) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/StatefulIcebergPageSourceProvider.java]: move this into the memoize?

Comment (jasonf20) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/StatefulIcebergPageSourceProvider.java]: done

**Pull request closed without merging**